iphone/cli/hooks/storyboard.js

```diff
@@ -0,0 +1,591 @@
+	 * Integrates a storyboards into the Xcode project by adding the required
+	 * build phases and adjusting the storyboard search path
+	 *
+	 * @param {storyboardInfo} storyboardInfo - storyboard metadata
+	 * @access public
+	 */
+	integrateStoryboard(storyboardInfo) {
+		const fileRefUuid = this.addStoryboardFileReference(storyboardInfo);
+		this.addLinkStoryboardBuildPhase(storyboardInfo, fileRefUuid);
+	}
+
+	/**
+	 * Add the storyboard as a new file reference to the Xcode project.
+	 *
+	 * @param {StoryboardInfo} storyboardInfo - storyboard metadata
+	 * @return {String} Uuid of the created file reference
+	 * @access private
+	 */
+	addStoryboardFileReference(storyboardInfo) {
+		const storyboardName = storyboardInfo.packageName;
+		const fileRefUuid = this._builder.generateXcodeUuid();
+		this._xobjs.PBXFileReference[fileRefUuid] = {
+			isa: 'PBXFileReference',
+			lastKnownFileType: 'file.storyboard',
+			path: `"${storyboardInfo.path}"`,
+			sourceTree: '"<group>"'
+		};
+		this._xobjs.PBXFileReference[fileRefUuid + '_comment'] = storyboardName;
+		this._resourceGroup.children.push({
+			value: fileRefUuid,
+			comment: storyboardName
+		});
+
+		return fileRefUuid;
+	}
+
+	/**
+	 * Adds the storyboard to the project's link storyboards build phase.
+	 *
+	 * @param {StoryboardInfo} storyboardInfo - storyboard metadata
+	 * @param {String} fileRefUuid - Uuid of the storyboards file reference inside the Xcode project
+	 * @access private
+	 */
+	addLinkStoryboardBuildPhase(storyboardInfo, fileRefUuid) {
+		const storyboardName = storyboardInfo.packageName;
+		const buildFileUuid = this._builder.generateXcodeUuid();
+		const buildFile = {
+			isa: 'PBXBuildFile',
+			fileRef: fileRefUuid,
+			fileRef_comment: storyboardName
+		};
+
+		this._xobjs.PBXBuildFile[buildFileUuid] = buildFile;
+		this._xobjs.PBXBuildFile[buildFileUuid + '_comment'] = storyboardName + ' in Resources';
+		this._resourceBuildPhase.files.push({
+			value: buildFileUuid,
+			comment: storyboardName + ' in Resources'
+		});
+	}
+}
+
+/**
+ * Inspects a storyboard and collects data about it that is required to integrate
+ * it with the Xcode project.
+ *
+ * The storyboard metadata that is collected here will also be added to the
+ * builder at the end of this hook so it can then be used by other hooks.
+ */
+class StoryboardInspector {
+
+	/**
+	 * Constructs a new storyboard inspector.
+	 *
+	 * @param {Object} logger - Appc logger instance
+	 * @access public
+	 */
+	constructor(logger) {
+		this._logger = logger;
+	}
+
+	/**
+	 * Inspects the storyboard under the given path and returns a new {@link StoryboardInfo}
+	 * instance for it.
+	 *
+	 * @param {String} storyboardPath - Path to the storyboard to inspect
+	 * @return {Promise<StoryboardInfo>}
+	 * @access public
+	 */
+	async inspect(storyboardPath) {
+		const packageExtension = path.extname(storyboardPath);
+		const storyboardName = path.basename(storyboardPath, packageExtension);
+		let storyboardInfo;
```

Comment:
why define `storyboardInfo` variable here? Why not just do a `return new StoryboardInfo(meta);` at the bottom?

Reply:
done.